core/src/avm2/globals/flash/display/display_object_container.rs

```diff
@@ -245,8 +245,8 @@ pub fn remove_child<'gc>(
             .get(0)
             .cloned()
             .unwrap_or(Value::Undefined)
-            .as_object()
-            .and_then(|o| o.as_display_object())
+            .coerce_to_object_or_typeerror(activation, None)?
```

Comment:
FYI, FP throws a different error here:
```
TypeError: Error #2007: Parameter child must be non-null.
```
While `coerce_to_object_or_typeerror` is for a more generic error handling, this error 2007 is more of checking arguments.
Avmplus uses it like:
```
toplevel()->checkNull(charSet, "charSet");
```
so I'm guessing we should have some separate method for it too?

That said, don't think the exact message is a priority, code 2007 is more important. We can also TODO it ofc.

Reply:
Fixed!